aztec-nargo/compile_then_postprocess.sh

```diff
@@ -57,11 +65,12 @@ for artifact in $artifacts_to_process; do
 
   # Now, update the artifact sequentially with each generated verification key.
   for fn_index in $private_fn_indices; do
-    vk_tmp="$artifact.verification_key_$fn_index.tmp"
+    fn_artifact=$(jq -r ".functions[$fn_index]" "$artifact")
+    fn_artifact_hash=$(echo "$fn_artifact" | sha256sum | cut -d' ' -f1)
+    vk_tmp="$artifact.verification_key_$fn_artifact_hash.tmp"
     verification_key=$(cat "$vk_tmp")
     # Update the artifact with the new verification key.
-    jq ".functions[$fn_index].verification_key = \"$verification_key\"" "$artifact" > "$artifact.tmp"
+    jq ".functions[$fn_index].verification_key = \"$verification_key\"" "$artifact" >"$artifact.tmp"
```

Comment:
my formatter kicked in.

To revert:
```suggestion
    jq ".functions[$fn_index].verification_key = \"$verification_key\"" "$artifact" > "$artifact.tmp"
```